### PR TITLE
fix(ctb): Clean up the preimage oracle

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Cleans up the PreimageOracle contract.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6269** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6269" target="_blank">Graphite</a> 
    * **PR #6270** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6270" target="_blank">Graphite</a>  👈
      * **PR #6276** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6276" target="_blank">Graphite</a> 
        * **PR #6293** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6293" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6270?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/cannon/PreimageOracle.sol [hunk 1]
```diff
@@ -37,17 +53,46 @@ contract PreimageOracle {
         preimageLengths[key] = size;
     }
 
-    // loadKeccak256PreimagePart prepares the pre-image to be read by keccak256 key,
-    // starting at the given offset, up to 32 bytes (clipped at preimage length, if out of data).
-    function loadKeccak256PreimagePart(uint256 partOffset, bytes calldata preimage) external {
+    /// @notice Computes and returns the key for a pre-image.
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/cannon/PreimageOracle.sol:56]: why no natspec treatment on `cheat()` above?

Comment (maurelian) [reply packages/contracts-bedrock/contracts/cannon/PreimageOracle.sol:56]: >     // This method is DANGEROUS. And NOT FOR PRODUCTION.

Ack.

File: packages/contracts-bedrock/contracts/cannon/PreimageOracle.sol [hunk 2]
```diff
@@ -1,24 +1,40 @@
...
     mapping(bytes32 => uint256) public preimageLengths;
+
+    /// @notice Mapping of pre-image keys to pre-image parts.
     mapping(bytes32 => mapping(uint256 => bytes32)) public preimageParts;
+
+    /// @notice Mapping of pre-image keys to pre-image part offsets.
     mapping(bytes32 => mapping(uint256 => bool)) public preimagePartOk;
 
-    function readPreimage(bytes32 key, uint256 offset)
+    /// @notice Reads a preimage from the oracle.
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/cannon/PreimageOracle.sol]: nit: inconsistent use of pre-image or preimage.

**Review by protolambda — APPROVED** [review]

**Review by clabby — APPROVED** [review]

**Comment by clabby** [comment]
@mergifyio queue

**Pull request merged**